src/xlat-i/xi-Parameter.C

```diff
@@ -826,15 +998,19 @@ void ParamList::unmarshall(XStr& str, bool isInline, bool isFirst, bool isRdmaPo
   else if (!isVoid()) {
     bool isSDAGGen = entry->sdagCon || entry->isWhenEntry;
     if (param->isRdma()) {
-      str << "\n#if CMK_ONESIDED_IMPL\n";
-      str << "ncpyBuffer_" << param->getName() << "_ptr";
-      str << "\n#else\n";
-      // for recv rdma
-      if(param->isRecvRdma())
+      if (param->isDevice()) {
+        str << "deviceBuffer_" << param->getName() << "_ptr";
+      } else {
```

Comment:
I think this code is exercised by the code path when CMK_ONESIDED_IMPL is 0 i.e. the ZC CPU arrays are copied as regular arrays, so I think the device code will not reach here. Can you confirm and remove the device if branch?

Reply:
This code is actually under `ParamList::unmarshall`, I think you mistook it for `Parameter::unmarshallRegArrayData`?

Reply:
Ah yes, that's exactly what happened.